app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,77 @@
+# TASKS = [
+#   'read POODR',
+#   'grocery shopping',
+#   'do laundry'
+# ]
+
+class TasksController < ApplicationController
+
+  def index
+    # @tasks = TASKS
+    @tasks = Task.all
+  end
+
+  def show
+    task_id = params[:id].to_i
+    @task = Task.find_by(id: task_id)
+    if @task.nil?
+      head :not_found
+    end
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    @task = Task.new(task_params)
+
+    if @task.save
+      redirect_to tasks_path
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @task = Task.find_by(id: params[:id])
+  end
```

Comment:
What happens if the task isn't found? You should include error-handling code in all of these actions that look for a single task.